### PR TITLE
Publish Code Coverage Results: refresh and remove PowerShell

### PR DESCRIPTION
**Pull request by DavidStaheli**
This PR:
1. Refreshes the task code.
2. Removes the PowerShell implementation.
3. Allows the "Summary File" and "Report Directory" inputs to include minimatch patterns to address issue #3690.

### CODE REVIEW

File: Tasks/PublishCodeCoverageResults/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
-  "loc.description": "Publish code coverage results to VSTS/TFS",
-  "loc.instanceNameFormat": "Publish Code Coverage Results $(summaryFileLocation)",
+  "loc.description": "Publish Cobertura or JaCoCo code coverage results from a build",
+  "loc.instanceNameFormat": "Publish code coverage from $(summaryFileLocation)",
   "loc.input.label.codeCoverageTool": "Code Coverage Tool",
-  "loc.input.help.codeCoverageTool": "Select the tool with which code coverage summary files have been generated",
+  "loc.input.help.codeCoverageTool": "Select the tool with which code coverage results are generated.",
   "loc.input.label.summaryFileLocation": "Summary File",
-  "loc.input.help.summaryFileLocation": "Path of code coverage summary file, which has code coverage statistics like line, method, class coverage. For example, $(Build.SourcesDirectory)/CodeCoverage/Summary.xml",
+  "loc.input.help.summaryFileLocation": "Enter the path of the summary file containing code coverage statistics, such as line, method, and class coverage. The value may contain minimatch patterns. For example, `$(System.DefaultWorkingDirectory)/MyApp/**/site/cobertura/coverage.xml`",
```
Comment (lkillgore) [inline Tasks/PublishCodeCoverageResults/Strings/resources.resjson/en-US/resources.resjson]: In most other Tasks we use the passive / descriptive form over the active form:
"Path of the summary file..."
vs.
"Enter the path..."

How important is consistency?

Comment (DavidStaheli) [inline Tasks/PublishCodeCoverageResults/Strings/resources.resjson/en-US/resources.resjson]: I've seen a mix, but you make me think shorter is better.

File: Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts [hunk 2]
```diff
@@ -1,23 +1,86 @@
...
+        if (pathMatches.length == 0) {
+            resolvedPath = undefined;
+        }
+        else {
+            // Select the path to be used from the matches
+            resolvedPath = pathMatches[0];
+
+            // If more than one path matches, use the first and issue a warning
+            if (pathMatches.length > 1) {
+                tl.warning(tl.loc('MultipleSummaryFilesFound', resolvedPath));
```
Comment (lkillgore) [inline Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: This is called for both a file and a directory.  This is the error message: "Multiple matching code coverage summary files were found. Using the first match: %s".  Should we change this to be more generic for files and folders?

Comment (DavidStaheli) [inline Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: Good catch!  Done.

**Pull request merged**